tls/s2n_signature_algorithms.c

```diff
@@ -50,13 +52,28 @@ static int s2n_signature_scheme_valid_to_offer(struct s2n_connection *conn, cons
 static int s2n_signature_scheme_valid_to_accept(struct s2n_connection *conn, const struct s2n_signature_scheme *scheme)
 {
     POSIX_ENSURE_REF(scheme);
+    POSIX_ENSURE_REF(conn);
 
     POSIX_GUARD(s2n_signature_scheme_valid_to_offer(conn, scheme));
 
     if (scheme->maximum_protocol_version != S2N_UNKNOWN_PROTOCOL_VERSION) {
         POSIX_ENSURE_LTE(conn->actual_protocol_version, scheme->maximum_protocol_version);
     }
 
+    if (conn->actual_protocol_version >= S2N_TLS13
+            || conn->actual_protocol_version == S2N_UNKNOWN_PROTOCOL_VERSION) {
+        POSIX_ENSURE_NE(scheme->hash_alg, S2N_HASH_SHA1);
+        POSIX_ENSURE_NE(scheme->sig_alg, S2N_SIGNATURE_RSA);
+        if (scheme->sig_alg == S2N_SIGNATURE_ECDSA) {
+            POSIX_ENSURE_REF(scheme->signature_curve);
+        }
+    }
+
+    if (conn->actual_protocol_version < S2N_TLS13) {
+        POSIX_ENSURE_EQ(scheme->signature_curve, NULL);
```

Comment:
I'm not sure we need the signature_curve checks. Those have more to do with properly constructing signature schemes than with properly choosing them, so are probably fine as unit tests. I also assume we ENSURE_REF the curve before we use it later.

Reply:
That is true -- I will remove line 73.